internal/rpmmd/repository.go

```diff
@@ -392,10 +392,9 @@ func NewRPMMD(cacheDir, dnfJsonPath string) RPMMD {
 	}
 }
 
-func (repo RepoConfig) toDNFRepoConfig(rpmmd *rpmmdImpl, i int, arch, releasever string) (dnfRepoConfig, error) {
-	id := strconv.Itoa(i)
+func (repo RepoConfig) toDNFRepoConfig(rpmmd *rpmmdImpl, arch, releasever string) (dnfRepoConfig, error) {
 	dnfRepo := dnfRepoConfig{
-		ID:             id,
+		ID:             repo.Name,
```

Comment:
Isn't it a bit misleading that the "name" is assigned to "id" even though RPM repos have "name" option as well? Isn't the RPM repository name used in the error output in case it is available?

Reply:
Good, point. I'll investigate this option further.